File: Itido-BMS/RentAPI/Rent/Models/Projects/ProjectItemAccess.cs
using System;
using Rent.Models.TimePlanning;

namespace Rent.Models.Projects
{
    public class ProjectItemAccess
    {
        public int ProjectItemID { get; set; }
        public int ProjectRoleID { get; set; }
        public bool Read { get; set; }
        public bool Write { get; set; }
        public virtual ProjectItem ProjectItem { get; set; }
        public virtual ProjectRole ProjectRole { get; set; }
    }
}

File: Itido-BMS/dart/bms_dart/lib/src/models/project_item_access.dart
class ProjectItemAccess {
  int projectItemID;
  int projectRoleID;
  bool read;
  bool write;

  ProjectItemAccess({
    this.projectItemID,
    this.projectRoleID,
    this.read,
    this.write,
  });
  ProjectItemAccess.fromJson(json)
      : this.projectItemID = json['projectItemID'],
        this.projectRoleID = json['projectRoleID'],
        this.read = json['read'],
        this.write = json['write'];

  Map<String, dynamic> toMap() => {
        'projectItemID': this.projectItemID,
        'projectRoleID': this.projectRoleID,
        'read': this.read,
        'write': this.write,
      };
}
